lib/Sema/CSSyntacticElement.cpp

```diff
@@ -1769,7 +1775,9 @@ class SyntacticElementSolutionApplication
       // statement that requires both source and destination
       // `case` preambles to be type-checked, so bodies of `case`
       // statements should be visited after preambles.
-      visitCaseStmtPreamble(caseStmt);
+      if (visitCaseStmtPreamble(caseStmt)) {
+        limitExhaustivityChecks = true;
```

Comment:
I believe we should already limit exhaustivity checks if we had an error, so I don't believe this should be neeeded

Reply:
We do in `TypeCheckStmt.cpp`, but not here, so stuff like [this](https://github.com/apple/swift/pull/63992/files#diff-e263c0ada91c06cf545151c0e6cf9ac9b881e9709936b53bd5f2eefc460317f0R1460) will currently crash.

Reply:
Ah, presumably you added this before I merged 6a7878b90de95b257fe285e958aebee5d27cf1d9 :)

Reply:
Yeah, that must be it! Would relying on the overall `hadError` not limit exhaustivity checks if something completely unrelated failed though, e.g. a neighboring syntactic element or something in the body of a case statement? I did this rather than `|| hadError` so that only errors in the preambles contribute.

Reply:
In theory, yes, but in practice the cases where we fail CSApply should be pretty limited at this point (and even more so for patterns once we land #63963), and eventually CSApply ought to never fail (either the solver should reject something, or it should just be a diagnostic that recovers). That being said, I'm happy to keep your version of the logic for now (and remove the `|| hadError` check).

Reply:
Let’s keep it if I come up with a test case then.